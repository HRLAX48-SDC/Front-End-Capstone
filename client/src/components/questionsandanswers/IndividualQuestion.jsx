import React, { useState, useEffect } from 'react';
import Answers from './Answers.jsx';
import { Button, Stack, Container, Row, Col } from 'react-bootstrap';
import axios from 'axios';
import API_KEY from '../../config/config.js';

const IndividualQuestion = (props) => {
  const question_id = props.question.question_id;

  const [q_helpful_count, setQCount] = useState(() => {
    return props.question.question_helpfulness;
  });
  const [markHelp, setMark] = useState(false);
  const markHelpful = () => {
    axios
      .put(`http://54.219.72.194/qa/questions/${question_id}/helpful`)
      .then((response) => {
        setQCount(q_helpful_count + 1);
        setMark(true);
      })
      .catch((err) => console.error(err));
  };

  const [reported, setReported] = useState(false);
  const markReport = () => {
    axios
      .put(`http://54.219.72.194/qa/questions/${question_id}/report`)
      .then((response) => {
        setReported(true);
      })
      .catch((err) => console.error(err));
  };

  return (
    <Container className='q_individual_q_grid' fluid='md'>
      <Row id='q_grid_row1'>
        <Col id='q_grid_row1_col1' xs={1}>
          <strong className='q_title'>Q:</strong>
        </Col>
        <Col id='q_grid_row1_col2'>
          <strong className='q_body'>{props.question.question_body}</strong>
        </Col>
        <Col id='q_grid_row1_col3' md={4}>
          <Stack direction='horizontal' gap={2}>
            <div className='ms-auto'>
              {!markHelp ? (
                <div>
                  <label>Helpful? </label>
                  <u id='q_helpful' onClick={() => markHelpful()}>
                    Yes
                  </u>
                  <span>({q_helpful_count})</span>
                </div>
              ) : (
                <div className='q_help_afterclick'>
                  <label>Helpful?</label>
                  <label>Yes({q_helpful_count})</label>
                </div>
              )}
            </div>
            <div className='vr' />
            <div>
              {!reported ? (
                <u id='q_report' onClick={() => markReport()}>
                  Report
                </u>
              ) : (
                <label className='q_reported'>Reported</label>
              )}
            </div>
          </Stack>
        </Col>
      </Row>
      <Answers question={props.question} product_name={props.product_name} />
      <br></br>
    </Container>
  );
};

export default IndividualQuestion;
