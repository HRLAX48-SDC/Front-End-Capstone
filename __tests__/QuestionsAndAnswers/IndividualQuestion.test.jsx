import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import IndividualQuestion from '../../client/src/components/questionsandanswers/IndividualQuestion.jsx';
Enzyme.configure({ adapter: new Adapter()});

const question = {
  question_id: 426448,
  question_body: 'Is it water proof?',
  question_date: "2021-09-24T00:00:00.000Z",
  question_helpfulness: 6,
  asker_name: "username123",
  answers: []
};
test('Render correctly', () => {
  const Question = renderer.create(<IndividualQuestion question={question} product_name='Camo Onesie' />).toJSON();
  expect(Question).toMatchSnapshot();
})
test('Render the question component', () => {
  const wrapper = shallow(<IndividualQuestion question={question} product_name='Camo Onesie' />);
  expect(wrapper.find('.q_title').text()).toEqual('Q:');
  expect(wrapper.find('#q_grid_row1_col3')).toContainMatchingElement('span');
  expect(wrapper).toContainExactlyOneMatchingElement('Answers');
});
