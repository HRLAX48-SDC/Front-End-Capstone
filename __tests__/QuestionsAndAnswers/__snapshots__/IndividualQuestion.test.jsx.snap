// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly 1`] = `
<div
  className="q_individual_q_grid container-md"
>
  <div
    className="row"
    id="q_grid_row1"
  >
    <div
      className="col-1"
      id="q_grid_row1_col1"
    >
      <strong
        className="q_title"
      >
        Q:
      </strong>
    </div>
    <div
      className="col"
      id="q_grid_row1_col2"
    >
      <strong
        className="q_body"
      >
        Is it water proof?
      </strong>
    </div>
    <div
      className="col-md-4"
      id="q_grid_row1_col3"
    >
      <div
        className="hstack gap-2"
      >
        <div
          className="ms-auto"
        >
          <div>
            <label>
              Helpful? 
            </label>
            <u
              id="q_helpful"
              onClick={[Function]}
            >
              Yes
            </u>
            <span>
              (
              6
              )
            </span>
          </div>
        </div>
        <div
          className="vr"
        />
        <div>
          <u
            id="q_report"
            onClick={[Function]}
          >
            Report
          </u>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="row"
      id="q_grid_row2"
    >
      <div
        className="col-1"
        id="q_grid_row2_col1"
      >
        <strong>
          A:
        </strong>
      </div>
      <div
        className="q_answerslist col"
        id="q_grid_row2_col2"
      >
        <div
          className="col"
        >
          <div
            className="hstack gap-2"
          >
            <div
              id="add_answer_button"
            >
               
              <button
                className="answer_add btn btn-outline-secondary btn-sm"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                ADD A ANSWER
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <br />
</div>
`;
